hoverxref/extension.py

```diff
@@ -139,6 +140,66 @@ def setup_sphinx_tabs(app, config):
             app.disconnect(listener_id)
 
 
+def setup_intersphinx(app, config):
+    """
+    Disconnect ``missing-reference`` from ``sphinx.ext.intershinx``.
+
+    As there is no way to hook into the ``missing_referece`` function to add
+    some extra data to the docutils node returned by this function, we
+    disconnect the original listener and add our custom one.
+
+    https://github.com/sphinx-doc/sphinx/blob/53c1dff/sphinx/ext/intersphinx.py
+    """
+
+    hoverxref_intersphinx_enabled = app.config.hoverxref_intersphinx or app.config.hoverxref_auto_ref
```

Comment:
`hoverxref_intersphinx` should default to `None` here. That way we can have `hoverxref_auto_ref=True` and `hoverxref_intersphinx=False` to enable hoverxref in all of them but disable it on intersphinx.